Chartboost/src/main/java/com/mopub/mobileads/ChartboostAdapterConfiguration.java

```diff
@@ -77,7 +77,7 @@ public void initializeNetwork(@NonNull Context context, @Nullable Map<String, St
             try {
                 if (configuration != null && !configuration.isEmpty()) {
 
-                    ChartboostShared.initializeSdk(context, configuration);
+                    boolean isInitialized = ChartboostShared.initializeSdk(context, configuration);
```

Comment:
Sorry remind me again, can SDK initialization fail if appID and appSignature are valid? It doesn't matter what config result we get right? Initialization always succeeds as long as appID and sig are okay?

Reply:
I'm asking because this function seems to return true regardless of what the initialization result is, which will prevent the initialization retry below

Reply:
ChartboostShared can return false in some conditions. I tried to do minimal changes on the mediation but to be honest the whole initialization part in the configuration seems unnecessary. I need to ask mopub why is this done this way.

Reply:
The only condition I see it returns false is if the SDK is already initialized. So it would be redundant to call Chartboost.startWithAppId there too right?

I think everything from line 82 til 98 (until `networkInitializationSucceeded = true;` is redundant. ChartboostShared.initializeSdk() does it all already.
I we want to avoid initialization redundancy I think we should just remove that code.

I also would like to know about my first question. There's this check inside ChartboostShared were we assume that if we tried to initialize the SDK, then the SDK IS initialized, and it is not initialized again. This will break if initialization could fail.

Reply:
It feels redundant and Id like to remove it but I don't know the purpose it has been put the in the first place. 
You first question could also explain why the code is redundant in the first place. Maybe it has been done like this to handle the case where init failed and this redundant code was doing init again.

Reply:
If you answer my question maybe I can help you figure out the reason this is here, and come up with a solution that addresses the issue 🙂

Reply:
Ping me on slack if something's not clear, we can discuss on a call